vowpalwabbit/csoaa.cc

```diff
@@ -470,6 +470,23 @@ void do_actual_learning(ldf& data, base_learner& base)
   if (start_K > 0)
     for (size_t k=1; k<K; k++)
       LabelDict::del_example_namespaces_from_example(*data.ec_seq[k], *data.ec_seq[0], (data.all->audit || data.all->hash_inv));
+
```

Comment:
Ok.  I smell a bad factoring in our code, but that's beyond what you are 
trying to solve.

-John

On 10/28/2015 11:09 AM, Martin Popel wrote:
>
> In vowpalwabbit/csoaa.cc 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/837#discussion_r43268017>:
>
> > @@ -470,6 +470,23 @@ void do_actual_learning(ldf& data, base_learner& base)
> >    if (start_K > 0)
> >      for (size_t k=1; k<K; k++)
> >        LabelDict::del_example_namespaces_from_example(*data.ec_seq[k], *data.ec_seq[0], (data.all->audit || data.all->hash_inv));
> > +
>
> I need ldf, so I cannot use |--oaa| (yes, VW is missing |--oaa_ldf|, 
> but that's a minor problem). In |parse_args.cc| I check that the 
> command line is |--csoaa_ldf=mc --loss_function=logisitic|, so the 
> probabilities are well defined. See also my last comment in #566 
> <https://github.com/JohnLangford/vowpal_wabbit/issues/566> .
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/837/files#r43268017>.
>